opensearchpy/connection/http_async_requests.py

```diff
@@ -0,0 +1,287 @@
+            if verify_certs is VERIFY_CERTS_DEFAULT:
+                verify_certs = True
+            if ssl_show_warn is SSL_SHOW_WARN_DEFAULT:
+                ssl_show_warn = True
+
+            if verify_certs:
+                ssl_context.verify_mode = ssl.CERT_REQUIRED
+                ssl_context.check_hostname = True
+            else:
+                ssl_context.check_hostname = False
+                ssl_context.verify_mode = ssl.CERT_NONE
+
+            ca_certs = self.default_ca_certs() if ca_certs is None else ca_certs
+            if verify_certs:
+                if not ca_certs:
+                    raise ImproperlyConfigured(
+                        "Root certificates are missing for certificate "
+                        "validation. Either pass them in using the ca_certs parameter or "
+                        "install certifi to use it automatically."
+                    )
+                if os.path.isfile(ca_certs):
+                    ssl_context.load_verify_locations(cafile=ca_certs)
+                elif os.path.isdir(ca_certs):
+                    ssl_context.load_verify_locations(capath=ca_certs)
+                else:
+                    raise ImproperlyConfigured("ca_certs parameter is not a path")
+            else:
+                if ssl_show_warn:
+                    warnings.warn(
+                        "Connecting to %s using SSL with verify_certs=False is insecure."
+                        % self.host
+                    )
+
+            # Use client_cert and client_key variables for SSL certificate configuration.
+            if client_cert and not os.path.isfile(client_cert):
+                raise ImproperlyConfigured("client_cert is not a path to a file")
+            if client_key and not os.path.isfile(client_key):
+                raise ImproperlyConfigured("client_key is not a path to a file")
+            if client_cert and client_key:
+                ssl_context.load_cert_chain(client_cert, client_key)
+            elif client_cert:
+                ssl_context.load_cert_chain(client_cert)
+
+        self.headers.setdefault("connection", "keep-alive")
+        self.loop = loop
+        self.session = None
+
+        # Parameters for creating an aiohttp.ClientSession later.
+        self._limit = maxsize
+        self._http_auth = http_auth
+        self._ssl_context = ssl_context
+
+    async def perform_request(
+        self, method, url, params=None, body=None, timeout=None, ignore=(), headers=None
+    ):
+        if self.session is None:
+            await self._create_aiohttp_session()
+        assert self.session is not None
+        orig_body = body
+        url_path = self.url_prefix + url
+        if params:
+            query_string = urlencode(params)
+        else:
+            query_string = ""
+
+
+        # There is a bug in aiohttp that disables the re-use
+        # of the connection in the pool when method=HEAD.
+        # See: aio-libs/aiohttp#1769
```

Comment:
Nit: Make it a full link to make it clickable.

Reply:
Updated to full link.